chatsky/pipeline/pipeline/component.py

```diff
@@ -40,40 +40,56 @@ class PipelineComponent(abc.ABC, BaseModel, extra="forbid", arbitrary_types_allo
     """
     This class represents a pipeline component, which is a service or a service group.
     It contains some fields that they have in common.
-
-    :param before_handler: :py:class:`~.BeforeHandler`, associated with this component.
-    :type before_handler: Optional[:py:data:`~.ComponentExtraHandler`]
-    :param after_handler: :py:class:`~.AfterHandler`, associated with this component.
-    :type after_handler: Optional[:py:data:`~.ComponentExtraHandler`]
-    :param timeout: (for asynchronous only!) Maximum component execution time (in seconds),
-        if it exceeds this time, it is interrupted.
-    :param requested_async_flag: Requested asynchronous property;
-        if not defined, `calculated_async_flag` is used instead.
-    :param calculated_async_flag: Whether the component can be asynchronous or not
-        1) for :py:class:`~.pipeline.service.service.Service`: whether its `handler` is asynchronous or not,
-        2) for :py:class:`~.pipeline.service.group.ServiceGroup`: whether all its `services` are asynchronous or not.
-
-    :param start_condition: StartConditionCheckerFunction that is invoked before each component execution;
-        component is executed only if it returns `True`.
-    :type start_condition: Optional[:py:data:`~.StartConditionCheckerFunction`]
-    :param name: Component name (should be unique in single :py:class:`~.pipeline.service.group.ServiceGroup`),
-        should not be blank or contain `.` symbol.
-    :param path: Separated by dots path to component, is universally unique.
     """
 
     before_handler: BeforeHandler = Field(default_factory=BeforeHandler)
+    """
+    :py:class:`~.BeforeHandler`, associated with this component.
+    """
     after_handler: AfterHandler = Field(default_factory=AfterHandler)
+    """
+    :py:class:`~.AfterHandler`, associated with this component.
+    """
     timeout: Optional[float] = None
+    """
+    (for asynchronous only!) Maximum component execution time (in seconds),
+    if it exceeds this time, it is interrupted.
+    """
     requested_async_flag: Optional[bool] = None
+    """
+    Requested asynchronous property; if not defined,
+    `calculated_async_flag` is used instead.
+    """
     calculated_async_flag: bool = False
+    """
+    Whether the component can be asynchronous or not.
+
+    1) for :py:class:`~.pipeline.service.service.Service`: whether its `handler` is asynchronous or not,
+    2) for :py:class:`~.pipeline.service.group.ServiceGroup`: whether all its `services` are asynchronous or not.
+
+    """
     start_condition: StartConditionCheckerFunction = Field(default=always_start_condition)
+    """
+    StartConditionCheckerFunction that is invoked before each component execution;
```

Comment:
Could you go over pipeline docs and fix issues like missing links?

Reply:
You mean that StartConditionCheckerFunction here should be a link, and the same for all entities in the docs? Or are there other missing links?

Reply:
![image](https://github.com/user-attachments/assets/aafdb5a0-7cda-4ad5-93e9-943996d3dc08)
I think the field type annotation should already have a link to the `StartConditionCheckerFunction` (like in the screenshot with links to attachments).
If it is so, it's better to replace "StartConditionCheckerFunction" with "Function" in the docstring.

Otherwise "StartConditionCheckerFunction" should be turned into a link with `:py:data:`.

The issue I have with this line is that this is a reference to a library object yet it is in plain text so it looks like a regular word in the docs.